packages/styles/scss/components/link-with-icon/_link-with-icon.scss

```diff
@@ -47,15 +47,32 @@
       fill: currentColor;
     }
 
-    .#{$prefix}--link--disabled:visited {
-      // Carbon core's has ongoing discussion on disabled link support.
-      // Meanwhile, this CSS makes disabled and visited link makes the right color
-      color: $disabled-02;
+    &.#{$prefix}--link--disabled {
+      &:not([href]),
+      &:not([button]) {
```

Comment:
Question: Wondering what the change is here for. One guess I made is un-disabling link when `href` attribute is there, but `[button]` made me unsure about it. Any insights?

To put this topic in more context; Wondering if it affects CSS overrides scenario like what's described in: https://www.w3.org/TR/selectors-4/#zero-matches

Reply:
Thanks @asudoh, I agree. That's in the [current version of Carbon](https://github.com/carbon-design-system/carbon/blob/fd79e780e5c4c49a9a0720490d31cbfa641e8b6e/packages/components/src/components/link/_link.scss#L48-L53) we're using. I planned on removing once #2940 goes through.

Reply:
Makes sense, thank you very much for clarifying @kennylam! (My apologies that I didn’t see it came from Carbon)